aiohttp/web_urldispatcher.py

```diff
@@ -755,6 +756,9 @@ def routes(self):
     def named_resources(self):
         return MappingProxyType(self._named_resources)
 
+    def set_defaults(self, *, allow_head):
```

Comment:
The developer can choose to support head explicitly but nothing breaks if they don't.

Allowing HEAD to endpoints which act like they got a GET request and still write body doesn't break anything.

Both django and flask allow users to still (by default) write body even for HEAD requests. I'm not saying they're always right, but their solution is more correct than aiohttp currently.

Here's an example:

**app.py:**
```python
import asyncio
from aiohttp import web

async def handle(request):
    print('preparing slow response...')
    await asyncio.sleep(2, loop=request.app.loop)
    return web.Response(body=b'this is the body\n')

app = web.Application()
app.router.add_get('/', handle, allow_head=True)

web.run_app(app, host='localhost', port=8000)
```

**shell:**
```
~ ➤  curl -v http://localhost:8000
* Rebuilt URL to: http://localhost:8000/
*   Trying 127.0.0.1...
* Connected to localhost (127.0.0.1) port 8000 (#0)
> GET / HTTP/1.1
> Host: localhost:8000
> User-Agent: curl/7.47.0
> Accept: */*
> 
< HTTP/1.1 200 OK
< Content-Length: 17
< Content-Type: application/octet-stream
< Date: Wed, 22 Feb 2017 10:08:03 GMT
< Server: Python/3.5 aiohttp/2.0.0a0
< 
this is the body
* Connection #0 to host localhost left intact
~ ➤  curl -I http://localhost:8000
HTTP/1.1 200 OK
Content-Length: 17
Content-Type: application/octet-stream
Date: Wed, 22 Feb 2017 10:08:10 GMT
Server: Python/3.5 aiohttp/2.0.0a0

~ ➤  
```

Nothing went wrong here. The app could choose to notice the HEAD request and not write body, but I don't see any problem if those not bothered by this (fairly niche) optimisation don't bother with this.

**In summary:** your current approach explicitly contravenes the HTTP RFC. The issue you're worried about is an optimisation which will apply in a very small percentage of requests and where app authors already have all the tools they need to achieve the optimization.